ingresslibchart/templates/_util.tpl

```diff
@@ -0,0 +1,25 @@
+{{- /*
+ingresslibchart.util.merge will merge two YAML templates and output the result.
+This takes an array of three values:
+- the top context
+- the template name of the overrides (destination)
+- the template name of the base (source)
+*/}}
+{{- define "ingresslibchart.util.merge" -}}
+{{- $top := first . -}}
+{{- $overrides := fromYaml (include (index . 1) $top) | default (dict ) -}}
+{{- $tpl := fromYaml (include (index . 2) $top) | default (dict ) -}}
+{{- toYaml (merge $overrides $tpl) -}}
+{{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "helm.labels" -}}
```

Comment:
One idea is rather than calculating out labels yourself, you might be able to grab the labels from the "calling" chart. 

In the liquibase-migration chart, I was able to able to do this:
```
{{- $appName := .Values.migration.appName | default .Release.Name -}}
{{- $labelRef := .Values.migration.labelRef | default (printf "%s.labels" $appName) -}}

// insert other chart stuffs
  labels: {{- include $labelRef . | nindent 4 }}
```

That works because the services themselves define labels like ["buffer.labels"](https://github.com/broadinstitute/terra-helm/blob/master/charts/buffer/templates/ingress/ingress.yaml#L7), and the release [name](https://github.com/broadinstitute/terra-helmfile/blob/master/helmfile.yaml#L118) is set to the name from [versions.yaml](https://github.com/broadinstitute/terra-helmfile/blob/master/versions.yaml#L8). The usage of `default` and stuff is just so that we can override this if we need to--I have yet to find an inconsistency in the naming but I suppose it could happen.